### DISCUSSION TITLE
Want to hide closed milestone from project view

### DISCUSSION THREAD

**Discussion by EddisonShyi** [community/community#66996] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Projects/slicing by milestone

### Body

As time goes by, we have a lot of milestones. We want to use filter to show only opened milestones, in case of too much milestones show in the left side when using slice by milestones. 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by dagdynamik** [comment]
we have the same problem also.

**Comment by jaclynjessup** [comment]
This would be a great addition to the filter. Something like `-milestone: closed` currently we have to do `-milestone:` for each closed milestone

**Reply by faustas99** [reply]
any progress on this?

**Comment by lhein** [comment]
Same here. Why are closed milestones still displayed?

**Comment by brunerae** [comment]
This feature would be SUPER helpful for our planning.

**Reply by rajakumar94095-ux** [reply]
Raja kumar

**Comment by tamlyn10** [comment]
Yes this is super annoying.. something we need too

**Comment by tomasMizera** [comment]
This is something that would help us a lot too. So far we use filters like `updated:<@today-2w`, that then hides milestones with zero tasks (when splicing by `milestone`). It is rather a workaround, I'd like to see a filter like `-milestone:closed` instead

**Comment by geniusadapter** [comment]
+1
I would also like to have the same feature.

**Comment by anastis** [comment]
+1
I hope we can at least get an answer, no matter if it's positive or negative.

**Comment by ProNeverFake** [comment]
+1
This is definitely necessary :)

**Comment by davidterrell** [comment]
+1 Would be helpful to keep project views decluttered!

**Comment by midiakiasat** [comment]
This isn’t supported today. When slicing by milestone in Projects, GitHub does not distinguish between open and closed milestones, and there’s no `milestone:open` or `-milestone:closed` filter.

Current workarounds:

* Use `is:open` to hide closed issues (milestones remain visible).
* Filter by recency (e.g., `updated:>=@today-2w`) to implicitly hide inactive milestones.
* Maintain a separate “Active” project view and move completed milestone issues out of it.

At scale, the cleanest pattern is to treat milestones as temporary planning artifacts and remove completed ones from active project views.

A native `milestone:open` filter would be the proper solution.

**Comment by marmaestro** [comment]
"Upvoting" this by commenting. Indeed, would be super useful!

**Comment by ThomasBirkmaier** [comment]
Upvoting Comment Too, I'd love to see this implemented

**Comment by mroberts-vtpanda** [comment]
+1

**Comment by MahmoudSaleh998** [comment]
+1

**Comment by MahmoudSaleh998** [comment]
Here is a useful tip I am currently using: adding is:open to the filter automatically hides any milestones that no longer have active tasks, which makes perfect sense for keeping the view clean. 🚀

**Comment by rajakumar94095-ux** [comment]
https://github.com/orgs/community/discussions/66996#discussioncomment-18044547
